benchmark/tools/inference_workload_matgen.py

```diff
@@ -0,0 +1,423 @@
+    @click.option('--num-decode-nodes', type=int, required=True, help='Number of nodes for decode')
+    @click.option('--prefill-tp', type=int, default=1, help='Tensor parallelism for prefill')
+    @click.option('--prefill-pp', type=int, default=1, help='Pipeline parallelism for prefill')
+    @click.option('--prefill-cp', type=int, default=1, help='Communication parallelism for prefill')
+    @click.option('--decode-tp', type=int, default=1, help='Tensor parallelism for decode')
+    @click.option('--decode-pp', type=int, default=1, help='Pipeline parallelism for decode')
+    @click.option('--decode-cp', type=int, default=1, help='Communication parallelism for decode')
+    @click.option('--model', type=str, help='Name of predefined model')
+    @click.option('--hidden-size', type=int, help='Model hidden size (for custom model)')
+    @click.option('--num-layers', type=int, help='Number of model layers (for custom model)')
+    @click.option('--num-heads', type=int, help='Number of attention heads (for custom model)')
+    @click.option('--num-kv-heads', type=int, help='Number of KV attention heads (for custom model)')
+    @click.option('--dtype-size', type=int, help='Size of model dtype in bytes (for custom model)')
+    @click.option('--results-dir', type=str, help='Directory to save results')
+    @click.option('--rail-optimized/--no-rail-optimized', default=False, help='Whether to use rail optimization')
+    def generate(num_user_requests, batch_size, num_prefill_nodes, num_decode_nodes,
+                prefill_tp, prefill_pp, prefill_cp, decode_tp, decode_pp, decode_cp,
+                model, hidden_size, num_layers, num_heads, num_kv_heads, dtype_size,
+                results_dir, rail_optimized):
+        """Generate communication matrices for given configuration"""
+        
+        if model:
+            if model not in PREDEFINED_MODELS:
+                raise click.BadParameter(f"Unknown model {model}. Available models: {list(PREDEFINED_MODELS.keys())}")
+            model_config = PREDEFINED_MODELS[model]
+        else:
+            if not all([hidden_size, num_layers, num_heads, num_kv_heads, dtype_size]):
+                raise click.BadParameter("Must specify either --model or all custom model parameters")
+            model_config = ModelConfig(
+                hidden_size=hidden_size,
+                num_layers=num_layers,
+                num_heads=num_heads,
+                num_kv_heads=num_kv_heads,
+                dtype_size=dtype_size
+            )
+
+        world_size = num_prefill_nodes * prefill_tp + num_decode_nodes * decode_tp
+        # print(f"World size: {world_size}")
+        # print(f"Model config: {model_config}")
+
+        main(
+            num_user_requests=num_user_requests,
+            batch_size=batch_size,
+            num_prefill_gpus=num_prefill_nodes * 8,
+            num_decode_gpus=num_decode_nodes * 8,
```

Comment:
What does the multiplier 8 represent here? TP Dim or Num GPUs per Node?
It will be better to specify a constant here for clarity.